src/BuildQueue.cpp

```diff
@@ -220,30 +220,33 @@ static void ProcessThrottling(BuildQueue *queue)
     throttled = false;
 }
 
-BuildResult::Enum BuildQueueBuild(BuildQueue *queue)
+BuildResult::Enum BuildQueueBuild(BuildQueue *queue, MemAllocLinear* scratch)
 {
     // Make sure none of the build threads see in-progress state due to a spurious wakeup.
     MutexLock(&queue->m_Lock);
+    MutexLock(&queue->m_BuildFinishedMutex);
 
     // Initialize build queue with index range to build
     int32_t *build_queue = queue->m_Queue;
     RuntimeNode *runtime_nodes = queue->m_Config.m_RuntimeNodes;
 
     int amountQueued = 0;
+
     for (int i = 0; i < queue->m_Config.m_TotalRuntimeNodeCount; ++i)
     {
         RuntimeNode *runtime_node = runtime_nodes + i;
-
-        //to start up, let's enqueue all nodes that have 0 dependencies.
```

Comment:
I think it's worth leaving a comment that we're looking for cache hits top-down.

Reply:
Fixed